Change all sway references to swayfx

notpeelz opened the pull request:
##### Notes:

- I changed as little as possible to minimize merge conflicts with upstream
- `SWAYSOCK` wasn't renamed because it would break e.g python scripts using i3ipc
- Commands such as `swaynag_command` are unchanged
- The project description in the manpage (`sway(1)`) was updated, hopefully you will find it adequate :)
- `CONTRIBUTING.md` was updated to reflect the direction of this project
- Contributor and author mentions in manpages/documentation were updated

##### Packages:

AUR: I've tested the AUR packages and use the PKGBUILD on my own machine as a daily driver. I have sway installed side-by-side with swayfx with no conflicts.

RPM: I'm not familiar with RPM. I've updated the spec file to the best of my ability. Untested.

Nix: Unchanged. I don't have a Nix/NixOS environment to test with.

Fixes #191

notpeelz commented:
I'm thinking maybe it would be a good idea to rename `SWAYSOCK` to `SWAYFXSOCK` after all.
If a program calls `sway --get-socketpath`, it would return SwayFX's socket path. This would allow scenarios where sway is run as a nested compositor on top of SwayFX, without clobbering the environment variable. I've accidentally run sway with the default config at least a couple times... On Arch Linux, the default config sources `50-systemd-user.conf`, which updates `SWAYSOCK` in the systemd environment block, breaking all scripts that make use of `swaymsg`. Additionally, SwayFX's IPC might not remain 100% compatible with Sway's in the long run.

`SWAYSOCK` is pretty much the only thing remaining that could result in accidental "crosstalk" between compositors.

ErikReider commented on build-scripts/aur-git/50-systemd-user.conf:
```diff
@@ -12,5 +12,5 @@ exec systemctl --user import-environment DISPLAY \
      dbus-update-activation-environment --systemd DISPLAY \
                                                   SWAYSOCK \
-                                                  XDG_CURRENT_DESKTOP=sway \
+                                                  XDG_CURRENT_DESKTOP="swayfx:sway:wlroots" \
```
Does having it colon separated actually work? Will it be detected by the portals conf?

notpeelz replied:
Portals work fine on my end. If I initiate screen sharing from Firefox/Discord, it calls xdp-wlr. Everything else calls xdp-gtk (as defined in `swayfx-portals.conf`).

The colon-separated list is used by `NotShowIn=`/`OnlyShowIn=` in `.desktop` files: https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html

>  If `$XDG_CURRENT_DESKTOP` is set then it contains a colon-separated list of strings. In order, each string is considered. If a matching entry is found in `OnlyShowIn` then the desktop file is shown. If an entry is found in `NotShowIn` then the desktop file is not shown. If none of the strings match then the default action is taken (as above).
> `$XDG_CURRENT_DESKTOP` should have been set by the login manager, according to the value of the `DesktopNames` found in the session file. The entry in the session file has multiple values separated in the usual way: with a semicolon.

ErikReider commented on completions/bash/sway:
```diff
@@ -1,6 +1,6 @@
+# swayfx(1) completion
 
-_sway()
+_swayfx()
```
Shouldn't these file names be renamed to swayfx? (all completion files)

notpeelz replied:
I didn't rename any files to avoid creating unnecessary conflicts when pulling from upstream.
The files get renamed by meson at install time (see the changes in `meson.build`).
e.g `completions/bash/sway` gets deployed to `/usr/share/bash-completion/completions/swayfx`

ErikReider commented on completions/zsh/_sway:
```diff
@@ -1,17 +1,5 @@
-#compdef sway
-#------------
-# Description
-# -----------
-#
-# Completion script for the sway window manager (http://swaywm.org)
-#
-# ---------------------------------------------
-# Author
-# -------
-#
-# * Seth Barberee <seth.barberee@gmail.com>
-#
-# -------------------------------
```
We should keep these, wouldn't be nice to remove the cred from Seth :)

notpeelz replied:
Woops. I meant to get rid of the unnecessary comments. I'll keep the author mention.

ErikReider commented on completions/zsh/_swaymsg:
```diff
@@ -1,17 +1,4 @@
-# Author
-# --------
-#
-#  * Seth Barberee <seth.barberee@gmail.com>
```
Same here

ErikReider commented on meson.build:
```diff
@@ -273,19 +277,32 @@ if get_option('zsh-completions')
 		'completions/zsh/_sway',
 		'completions/zsh/_swaymsg',
 	)
+	swayfx_zsh_files = [
+		'_swayfx',
+		'_swayfxmsg',
+	]
```
Should this be here? Shouldn't `zsh_files` also be removed then?

notpeelz replied:
`swayfx_zsh_files` is passed to `install_data()` a couple lines below. It's used for renaming purposes.

ErikReider commented on meson.build:
```diff
@@ -273,19 +277,32 @@ if get_option('zsh-completions')
 		'completions/bash/sway',
 		'completions/bash/swaymsg',
 	)
+	swayfx_bash_files = [
+		'swayfx',
+		'swayfxmsg',
+	]
```
Same here

ErikReider commented on meson.build:
```diff
@@ -297,17 +314,26 @@ if get_option('bash-completions')
 		'completions/fish/sway.fish',
 		'completions/fish/swaymsg.fish',
 	)
+	swayfx_fish_files = [
+		'swayfx.fish',
+		'swayfxmsg.fish',
+	]
```
Same here

ErikReider commented on sway/sway.1.scd:
```diff
@@ -103,11 +102,11 @@ _XCURSOR_THEME_
-Maintained by Simon Ser <contact@emersion.fr>, who is assisted by other open
-source contributors. For more information about sway development, see
-<https://github.com/swaywm/sway>.
+Maintained by William McKinnon <contact@willmckinnon.com>, who is assisted by
```
@WillPower3309 Is this email in-use?

ErikReider reviewed (changes requested):
Some nits, hopefully the "same here" comments are in the correct order.

Kudos for rewriting some of the texts too! :)

notpeelz commented:
Marking as draft until the review is done.

*pokes @WillPower3309*

mochaaP commented:
Could we provide distro packages like `sway-is-swayfx` or `swayfx-sway` to provide `/usr/bin/sway{,msg,nag,bar}`? This could break some packages depending on the binaries or the `sway` package name.

notpeelz commented:
I've squashed the fixup commits and added a new subpackage to alias swayfx's versions of `sway{,bar,msg,nag}`.
I also tried modifying the Fedora spec file, but rpkg was giving me strange errors (tested in a Fedora 39 VM).
Also, [rpkg-util is no longer maintained/developed](https://pagure.io/rpkg-util), so we should probably use some other packaging tool (such as dist-git or tito).

mochaaP commented:
We usually use `fedpkg` for day-to-day packaging works.  <https://docs.fedoraproject.org/en-US/package-maintainers/Package_Maintenance_Guide/#common_fedpkg_commands> Hope this helps!

notpeelz commented:
Thanks! I'll look into it.

BTW, if we go forward with the `SWAYSOCK` rename, the compat packages could ship a config file to alias the environment variable, e.g.

`/etc/swayfx/config.d/50-swaysock-compat.conf`:
```
exec systemctl --user set-environment \
  SWAYSOCK="$SWAYFXSOCK" \
  I3SOCK="$SWAYFXSOCK"
```

notpeelz commented:
I rebased against the latest changes and got rid of most packaging-related commit.
This should make it easier to review. I plan on submitting the rest of the changes in separate PR(s).

WillPower3309 commented:
Hey peelz! Thanks a ton for all the work here and apologies on my end for the delays. I think it's a good call to save the swaysock change for a separate PR as I'd like to have more of a conversation there on if this is something we'd want

WillPower3309 commented on README.md:
```diff
@@ -44,10 +44,10 @@ Sway is an incredible window manager, and certainly one of the most well establi
     - `layer_effects <layer namespace> <effects>`
-    - The current layer namespaces can be shown with `swaymsg -r -t get_outputs | jq '.[0].layer_shell_surfaces | .[] | .namespace'`
+    - The current layer namespaces can be shown with `swayfxmsg -r -t get_outputs | jq '.[0].layer_shell_surfaces | .[] | .namespace'`
     - Example: `layer_effects "waybar" blur enable; shadows enable; corner_radius 6`
       - Note: If an application uses gtk, its namespace is likely to be "gtk-layer-shell"
-    - SwayIPC Example: `swaymsg "layer_effects 'waybar' 'blur enable; shadows enable; corner_radius 6'"`
+    - SwayIPC Example: `swayfxmsg "layer_effects 'waybar' 'blur enable; shadows enable; corner_radius 6'"`
```
not sure how I feel about this one. Do you mind if we save the IPC changes for a follow-up PR? I think we need some more discussion there

notpeelz replied:
That defeats the whole point of #191.

There's two options:
1. we close this PR and keep SwayFX as a drop-in replacement for Sway
2. we go forward with the rename and make SwayFX independent, allowing it to be installed side-by-side with Sway

If we go forward with the rename, we can't have any packaged files conflicting with upstream. 
Splitting up the changes into more PRs is only going to make it harder to complete the migration.

When it comes to "IPC compatibility", you can temporarily symlink the binaries, e.g.: `~/.local/bin/swaymsg` -> `/usr/bin/swayfxmsg`

As I stated in my earlier comments, I had started working on the packaging side of things. Before the rebase, I had a commit (a06a4ab409592161d398b0f8ca0a86ed86c975ed) that added a subpackage to create those symlinks. This would alleviate the friction when switching between SwayFX and Sway.

From what I understand both @RicArch97 and @WillPower3309 use Nix/NixOS, in which case you can create those symlinks in the `postInstall` step, i.e:

```nix
postInstall = (old.postInstall or "") + ''
  for f in "" msg nag bar; do
    ln -sT "swayfx$f" "$out/bin/sway$f"
  done
'';
```

WillPower3309 commented on build-scripts/aur-git/50-systemd-user.conf:
```diff
@@ -12,5 +12,5 @@ exec systemctl --user import-environment DISPLAY \
      dbus-update-activation-environment --systemd DISPLAY \
                                                   SWAYSOCK \
-                                                  XDG_CURRENT_DESKTOP=sway \
+                                                  XDG_CURRENT_DESKTOP="swayfx:sway" \
```
Q: does this env var play well with having multiple values?

notpeelz replied:
Yep. Erik asked the same question (see previous review).

WillPower3309 commented:
Just a minor comment and a question

werdahias commented:
Allowing side-by-side installing (where no binaries are named the same) with sway would also really ease packaging for debian and in consequence, ubuntu

sweiglbosker commented:
Why was this closed?

notpeelz commented:
There was some pushback and things weren't moving fast enough. Since I no longer use Sway/SwayFX, I decided to close it.

Feel free to pick it up if you want to see this merged.
